core-impl/client/src/main/java/com/alipay/sofa/rpc/client/ElasticConnectionHolder.java

```diff
@@ -0,0 +1,162 @@
+        if (LOGGER.isInfoEnabled(appName)) {
+            LOGGER.infoWithApp(appName, "Add provider of {}, size is : {}", interfaceId, providerSize);
+        }
+        if (providerSize > 0) {
+
+            int minSynConnectSize = 0;
+            //可自定义初始化连接的百分比数以及固定最小数
+            //计算初始化连接最少数,优先使用初始化最小数属性进行计算,百分比属性默认为0
+            if (elasticConnectPrecent > 0) {
+                double precent = elasticConnectPrecent >= 100 ? 1 : elasticConnectPrecent * 0.01;
+                minSynConnectSize = ((Double) (providerInfoList.size() * precent)).intValue();
+            } else {
+                minSynConnectSize = elasticConnectSize;
+            }
+
+            // 多线程建立连接
+            int threads = Math.min(10, minSynConnectSize); // 最大10个
+            final CountDownLatch latch = new CountDownLatch(minSynConnectSize);
+
+            int connectTimeout = consumerConfig.getConnectTimeout();
+
+            ThreadPoolExecutor initPool = new ThreadPoolExecutor(threads, threads,
+                0L, TimeUnit.MILLISECONDS,
+                new LinkedBlockingQueue<Runnable>(minSynConnectSize),
+                new NamedThreadFactory("CLI-CONN-" + interfaceId, true));
+
+            NamedThreadFactory namedThreadFactory = new NamedThreadFactory("CLI-ASYN-CONN-", true);
+            // 第一次同步建立连接的连接数
+            int synInitConnectProviderSize = 0;
+            for (final ProviderInfo providerInfo : providerInfoList) {
+                final ClientTransportConfig config = providerToClientConfig(providerInfo);
+                if (synInitConnectProviderSize >= minSynConnectSize) {
+                    break;
+                }
+                synInitConnectProviderSize++;
+                initClientRunnable(initPool, latch, providerInfo);
+            }
+
+            try {
+                int totalTimeout = ((synInitConnectProviderSize % threads == 0) ? (synInitConnectProviderSize / threads)
+                    : ((synInitConnectProviderSize /
+                    threads) + 1)) *
+                    connectTimeout + 500;
+                latch.await(totalTimeout, TimeUnit.MILLISECONDS); // 一直等到子线程都结束
+            } catch (InterruptedException e) {
+                LOGGER.errorWithApp(appName, "Exception when add provider", e);
+            } finally {
+                initPool.shutdown(); // 关闭线程池
+            }
+
+            final List<ProviderInfo> asynConnectProviderInfoList = providerInfoList.subList(synInitConnectProviderSize,
+                providerInfoList.size());
+
+            if (!asynConnectProviderInfoList.isEmpty()) {
+                if (LOGGER.isInfoEnabled(appName)) {
+                    LOGGER.infoWithApp(appName, "asynConnectProviderInfoListSize:{}",
+                        asynConnectProviderInfoList.size());
+                }
+                final ExecutorService executorService = Executors
+                    .newFixedThreadPool(asynConnectProviderInfoList.size());
```

Comment:
set `NamedThreadFactory` for executorService.

Reply:
Okay, I'll change it tonight